File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/firmware/GetLockFirmwareVersionCase.java
package com.lattis.ellipse.domain.interactor.lock.firmware;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 9/6/17.
 */

public class GetLockFirmwareVersionCase extends BaseLockUseCase<String> {

    private final String TAG = GetLockFirmwareVersionCase.class.getName();

    @Inject
    GetLockFirmwareVersionCase(ThreadExecutor threadExecutor,
                               PostExecutionThread postExecutionThread,
                               BluetoothRepository bluetoothRepository, LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread,bluetoothRepository,lockRepository);
    }


    public GetLockFirmwareVersionCase forLock(Lock lock) {
        setScannedLock(lock);
        return this;
    }

    @Override
    protected Observable<String> buildUseCaseObservable() {
        return getLockFirmwareVersion();
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/firmware/GetLockFirmwareVersionCase.kt
package com.lattis.domain.usecase.lock.firmware

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.models.Lock
import com.lattis.domain.repository.SaSOrPSLockRepository
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetLockFirmwareVersionCase @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository, lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<String>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository,
    lockRepository,
    saSOrPSLockRepository
) {
    private val TAG = GetLockFirmwareVersionCase::class.java.name
    fun forLock(lock: Lock): GetLockFirmwareVersionCase {
        setScannedLock(lock)
        return this
    }

    override fun buildUseCaseObservable(): Observable<String> {
        return lockFirmwareVersion
    }
}